CHANGELOG.md

```diff
@@ -57,3 +61,4 @@ Other minor additions
 
 * Made first argument of [,]-∘-distr in `Data.Sum.Properties` explicit
 * Added map-id, map₁₂-commute, [,]-cong, [-,]-cong, [,-]-cong, map-cong, map₁-cong and map₂-cong to `Data.Sum.Properties`
+* `Reflection.Pattern`: Calculation of number of bound variables in patterns with `pattern-size` and `pattern-args-size`
```

Comment:
Any chance you could add the types for these? Look at the CHANGELOG on master to give you an idea as to the format (omitting implicits or uninteresting variables)

Reply:
Sure. Also I wasn't sure whether those were the best names for those functions. Would `pattern-var-count` and `pattern-args-var-count` be better, or too verbose?

Reply:
Another point is that I don't fully understand why `pattern-size` is defined the way it is. It is a copy of a function in agda-prelude by https://github.com/UlfNorell/agda-prelude. I do not understand why `dot` has a size of one since as far as I can tell using `.(...)` in patterns does not introduce new variables, and I do not know what `proj` corresponds to in concrete syntax. It all seems to work though.

Reply:
Types have been added as requested, but I'm leaving this conversation unresolved to remind me to resolve my above comment.

Reply:
Pinging @UlfNorell?

Reply:
Counting the dot pattern is a bug in agda-prelude (thanks for finding!). It used to be that dot patterns counted as deBruijn indices, but this got fixed in agda/agda#2203.

Reply:
OK, thanks for clearing that up. I've corrected my copy of that function.